### PR TITLE
Introduce Scanamo library to reduce dynamoDB boilerplate.

### PR DESCRIPTION
**Pull request by LATaylor-guardian**

### CODE REVIEW

File: app/com/gu/floodgate/DynamoDBTable.scala [hunk 1]
```diff
@@ -40,25 +40,10 @@ trait DynamoDBTable[T] extends StrictLogging {
...
-      new QueryRequest().withTableName(tableName)
-        .withKeyConditionExpression(s"$keyName = :$keyName AND $sortKeyName = :$sortKeyName")
-        .withExpressionAttributeValues(Map(
-          s":$keyName" -> new AttributeValue().withS(hashKey),
-          s":$sortKeyName" -> new AttributeValue().withS(sortKey)
-        ))
-    }
-
-    maybeRequest.map(getThisItem(_)) getOrElse Future.successful(None)
+  def getItem[T: DynamoFormat](hashKey: String, sortKey: String) = {
```
Comment (cb372) [inline app/com/gu/floodgate/DynamoDBTable.scala]: Please add type annotations to all the public methods in this class

**Comment by cb372** [comment]
Nice! All hail @philwills

**Comment by LATaylor-guardian** [comment]
@philwills :clap:

File: app/com/gu/floodgate/jobhistory/JobHistoryService.scala [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+import com.gu.floodgate.reindex.ReindexStatus.{ Unknown }
+import com.gu.scanamo.DynamoFormat
+import org.joda.time.DateTime
 
 import scala.concurrent.Future
 
 class JobHistoryService(jobHistoryTable: DynamoDBTable[JobHistory]) {
 
-  def createJobHistory(jobHistory: JobHistory) = jobHistoryTable.saveItem(jobHistory)
+  implicit val dateFormat = DynamoFormat.xmap(DynamoFormat.stringFormat)(d => Validated.valid(new DateTime(d)))(_.toString)
```
Comment (philwills) [inline app/com/gu/floodgate/jobhistory/JobHistoryService.scala]: Be warned that I've just realised I can probably make that first parameter implicit, so the signature is likely to change in future.

**Comment by cb372** [comment]
:+1:

**Pull request merged**